### PR TITLE
Photoshop: make new publisher default

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Removed old Pyblish and switched to `Publisher`

## Description
Updated documentation, updated extension menu

## Additional info
Needs to redeploy extension!!!
Refactored to new host class to standardize.

## Testing notes:
1. try various publishing in PS
2. try to publish workfile with instances created in Pyblish

### CODE REVIEW

**Review by 64qam — APPROVED** [review]

File: openpype/hosts/photoshop/api/pipeline.py [hunk 1]
```diff
@@ -26,6 +33,147 @@
 INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
 
 
+class PhotoshopHost(HostBase, IWorkfileHost, ILoadHost):
+    name = "photoshop"
+
+    def __init__(self):
+        super(PhotoshopHost, self).__init__()
+
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/api/pipeline.py:41]:
```suggestion
```

**Review by iLLiCiTiT — CHANGES_REQUESTED** [review]
Few comments

File: openpype/hosts/photoshop/api/pipeline.py [hunk 2]
```diff
@@ -26,6 +33,144 @@
 INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
 
 
+class PhotoshopHost(HostBase, IWorkfileHost, ILoadHost):
+    name = "photoshop"
+
+    def install(self):
+        """Install Photoshop-specific functionality of avalon-core.
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/api/pipeline.py:40]:
```suggestion
        """Install Photoshop-specific functionality needed for integration.
```

File: openpype/hosts/photoshop/plugins/create/workfile_creator.py [hunk 3]
```diff
@@ -17,7 +16,7 @@ def get_instance_attr_defs(self):
         return []
 
     def collect_instances(self):
-        for instance_data in api.list_instances():
+        for instance_data in self.host.list_instances():
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/create/workfile_creator.py:19]: I would recommend using shared data if instances are stored at the same place (shared data across creators during collection phase). Check TrayPublisher for inspiration https://github.com/pypeclub/OpenPype/blob/develop/openpype/hosts/traypublisher/api/plugin.py#L20 .

Comment (kalisp) [reply openpype/hosts/photoshop/plugins/create/workfile_creator.py:19]: Why, `self.host` gets reinitialized?

Or you mean to cache value of  `list_instances`? I could create internal `_instances` list and add it to `list_instances` method.

Comment (iLLiCiTiT) [reply openpype/hosts/photoshop/plugins/create/workfile_creator.py:19]: Attribute `collection_shared_data` is shared dictionary across all creators so they don't have to collect exactly the same data individually.

File: openpype/hosts/photoshop/plugins/create/create_image.py [hunk 4]
```diff
@@ -19,7 +20,10 @@ class ImageCreator(Creator):
     description = "Image creator"
 
     def collect_instances(self):
-        for instance_data in api.list_instances():
+        instances = cache_and_get_instances(self,
+                                            "openpype.photoshop.instances",
+                                            self.host.list_instances)
+        for instance_data in instances:
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/create/create_image.py:26]:
```suggestion
        for instance_data in instances.values():
```

File: openpype/hosts/photoshop/plugins/create/create_image.py [hunk 5]
```diff
@@ -163,6 +165,12 @@ def _handle_legacy(self, instance_data):
...
-        return {"layer": "{layer}"}
+
+    def get_dynamic_data(self, variant, task_name, asset_doc,
+                         project_name, host_name, instance):
+        if not instance:
+            return {"layer": "{layer}"}
+        else:
+            layer_name = instance.get("layer_name")
+            if layer_name:
+                return {"layer": layer_name}
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/create/create_image.py:176]: There is no default output
```suggestion
        if instance is not None:
            layer_name = instance.get("layer_name")
            if layer_name:
                return {"layer": layer_name}
        return {"layer": "{layer}"}
```

File: openpype/hosts/photoshop/api/pipeline.py [hunk 6]
```diff
@@ -226,28 +292,16 @@ def containerise(
...
-    return "{}/{}/{}".format(project_name, asset_name, task_name)
+    Storing all instances as a list as legacy instances might be still present.
+    Args:
+        creator (Creator): Plugin which would like to get instances from host.
+    Returns:
+        List[]: list of all instances stored in metadata
+    """
+    shared_key = "openpype.photoshop.instances"
+    if shared_key not in creator.collection_shared_data:
+        creator.collection_shared_data[shared_key] = creator.list_instances()
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/api/pipeline.py:306]:
```suggestion
        creator.collection_shared_data[shared_key] = creator.host.list_instances()
```

**Review by iLLiCiTiT — APPROVED** [review]
Looks good. It would be good if @64qam would test it again (after my comments) but I think we didn't change anything crutial.

**Review by 64qam — APPROVED** [review]

**Pull request merged**